core/server/worker/src/main/java/alluxio/worker/block/BlockStore.java

```diff
@@ -35,12 +35,23 @@
    *
    * @param sessionId the id of the session to lock this block
    * @param blockId the id of the block to lock
-   * @return the lock id if the lock is acquired successfully
+   * @return the lock id (non-negative) if the lock is acquired successfully
    * @throws BlockDoesNotExistException if block id can not be found, for example, evicted already
    */
   long lockBlock(long sessionId, long blockId) throws BlockDoesNotExistException;
 
   /**
+   * Locks an existing block and guards subsequent reads on this block. If the lock fails, return
+   * {@link BlockLockManager#INVALID_LOCK_ID}.
+   *
+   * @param sessionId the id of the session to lock this block
+   * @param blockId the id of the block to lock
+   * @return the lock id (non-negative) that uniquely identifies the lock obtained or
+   *         {@link BlockLockManager#INVALID_LOCK_ID} if it failed to lock
+   */
+  long lockBlockNoException(long sessionId, long blockId);
```

Comment:
@calvinjia @apc999  I I deprecate lockBlock (the one with exception) and use lockBlockNoException everywhere, then i can only throw exception like LockBlockFailedException instead BlockDoesNotExistException like what we are doing now.  Similar argument is applied to unlockBlock. 

One common way to fix this in a language (e.g. google c++) that does not use exception is to have a status class the passes the failure information upwards.